### DISCUSSION TITLE
Unnecessary events?

### DISCUSSION THREAD

**Discussion by jansedlon** [edmundhung/conform#291] [General]
Hello @edmundhung !

I was debugging through conform and I noticed something.
On these lines [link](https://github.com/edmundhung/conform/blob/8f061e584e24820cec09882f9c3cb4ed140646b1/packages/conform-react/hooks.ts#L1162-L1165).

Why is it submitting error event if the Array of errors is empty?  I've got a bunch of input in my form and I'm debugging it because I can't figure out why Conform is not setting an error for my uncontrolled date picker, but that's a different discussion. I found this out and thought why it's reporting errors for fields that don't have any errors :)

**Comment by jansedlon** [comment]
Okay, so the unnecessary emitted events still applies but I absolutely randomly figured out why the uncontrolled date picker didn't work. Apparently, the hidden input for storing the selected date cannot have both `value` and `readOnly`. It somehow messes up things.

**Reply by edmundhung** [reply]
Thanks for sharing what you found! I can confirm the issue you mentioned. Conform utilizes Constraint Validation APIs such as [setCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setCustomValidity) to propagate the error messages at the moment. But [the element might be barred from constraint validation on certain conditions](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#barred-from-constraint-validation). For example, the browser does not report error from `disabled` or `readonly` form control elements which is the issue you had.

I understand this made integrating custom inputs much harder and I am already working on an improvement to remove this limitation soon :)

**Reply by jansedlon** [reply]
Thank you for the information! 😊
